src/rtc_pl031.rs

```diff
@@ -38,6 +38,44 @@ const AMBA_IDS: [u8; 8] = [0x31, 0x10, 0x04, 0x00, 0x0d, 0xf0, 0x05, 0xb1];
 const AMBA_ID_LOW: u16 = 0xFE0;
 const AMBA_ID_HIGH: u16 = 0xFFF;
 
+/// Defines a series of callbacks that are invoked in response to the occurrence of specific
+/// failure or missed events as part of the RTC operation (e.g., write to an invalid offset). The
+/// methods below can be implemented by a backend that keeps track of such events by incrementing
+/// metrics, logging messages, or any other action.
+///
+/// We're using a trait to avoid constraining the concrete characteristics of the backend in
+/// any way, enabling zero-cost abstractions and use case-specific implementations.
+// TODO: The events defined below are just some examples for now to validate the approach. If
+// things look good, we can move on to establishing the initial list. It's also worth mentioning
+// the methods can have extra parameters that provide additional information about the event.
```

Comment:
I think we can remove this TODO. The PR looks ready to merge to me.

Reply:
Done.  Just pushed an update.